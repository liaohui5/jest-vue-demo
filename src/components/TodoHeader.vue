<template>
  <div class="header-wrapper">
    <div class="header-content">
      <span>Todo List</span>
      <input
        type="text"
        data-jest="input"
        placeholder="add todo item"
        v-model="inputVal"
        @keyup.enter="addTodo"
      />
    </div>
  </div>
</template>

<script>
export default {
  data: () => ({
    inputVal: ""
  }),
  methods: {
    addTodo() {
      if (!this.inputVal) return;
      this.$emit("add-todo", this.inputVal);
      this.inputVal = "";
    }
  }
};
</script>

<style lang="less" scoped>
.header-wrapper {
  width: 100%;
  line-height: 80px;
  background: #333;
  .header-content {
    width: 600px;
    margin: 0 auto;
    color: #fff;
    font-size: 24px;
    input {
      float: right;
      width: 360px;
      margin-top: 25px;
      line-height: 24px;
      outline: none;
      text-indent: 10px;
    }
  }
}
</style>
