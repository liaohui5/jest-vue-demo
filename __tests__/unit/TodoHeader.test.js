import TodoHeader from "../../src/components/TodoHeader.vue";
import { shallowMount } from "@vue/test-utils";

describe("TodoHeader.vue", () => {
  let wrapper;
  let input;

  beforeEach(() => {
    wrapper = shallowMount(TodoHeader);
    input = wrapper.find("input[data-jest='input']");
  });

  it("TodoHeader dom变化的时候, 提示确认更新", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("TodoHeader 应该包含 input", () => {
    expect(input.exists()).toBe(true);
  });

  it("TodoHeader 中 input 默认值为空字符串", () => {
    expect(wrapper.vm.$data.inputVal).toBe("");
  });

  it("TodoHeader 中 input 双向数据绑定", () => {
    const val = "hello";
    input.setValue(val);
    expect(wrapper.vm.$data.inputVal).toBe(val);
  });

  it("TodoHeader 中 input 内容为空时按下enter, 不能向外触发 add-todo 事件", () => {
    input.setValue("");
    input.trigger("keyup.enter");
    const emits = wrapper.emitted();
    expect(emits["add-todo"]).toBeFalsy();
  });

  it("TodoHeader 中 input 内容不为空时, 向外触发 add-todo 事件, 并且输入input", () => {
    input.setValue("todo-1");
    input.trigger("keyup.enter");
    const emits = wrapper.emitted();
    expect(emits["add-todo"]).toBeTruthy();
    expect(wrapper.vm.$data.inputVal).toBe("");
  });
});
