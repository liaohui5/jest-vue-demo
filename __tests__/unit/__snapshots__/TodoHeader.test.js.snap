// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoHeader.vue TodoHeader dom变化的时候, 提示确认更新 1`] = `
<div class="header-wrapper">
  <div class="header-content"><span>Todo List</span> <input type="text" data-jest="input" placeholder="add todo item"></div>
</div>
`;
